deeppavlov/configs/intents/intents_dstc2.json

```diff
@@ -22,65 +22,86 @@
     ]
   },
   "chainer": {
-    "in": ["x"],
-    "in_y": ["y"],
+    "in": [
+      "x"
+    ],
+    "in_y": [
+      "y"
+    ],
     "pipe": [
       {
         "id": "classes_vocab",
         "name": "default_vocab",
-        "fit_on": ["y"],
+        "fit_on": [
+          "y"
+        ],
         "level": "token",
         "save_path": "vocabs/classes.dict",
         "load_path": "vocabs/classes.dict"
       },
       {
-        "in": ["x"],
-        "in_y": ["y"],
-        "out": ["y_predicted"],
+        "id": "my_embedder",
+        "name": "fasttext",
+        "save_path": "embeddings/dstc2_fastText_model.bin",
+        "load_path": "embeddings/dstc2_fastText_model.bin",
+        "dim": 100
+      },
+      {
+        "id": "my_tokenizer",
+        "name": "nltk_tokenizer",
+        "tokenizer": "wordpunct_tokenize"
+      },
+      {
+        "in": [
+          "x"
+        ],
+        "in_y": [
+          "y"
+        ],
+        "out": [
+          "y_labels",
+          "y_probas",
+          "y_classes"
+        ],
         "main": true,
         "name": "intent_model",
-        "save_path": "intents/intent_cnn_v2",
-        "load_path": "intents/intent_cnn_v2",
+        "save_path": "intents/intent_cnn_v4",
+        "load_path": "intents/intent_cnn_v4",
         "classes": "#classes_vocab.keys()",
-        "opt": {
-          "kernel_sizes_cnn": [
-            3,
-            3,
-            3
-          ],
-          "filters_cnn": 512,
-          "confident_threshold": 0.5,
-          "optimizer": "Adam",
-          "lear_rate": 0.1,
-          "lear_rate_decay": 0.1,
-          "loss": "binary_crossentropy",
-          "text_size": 15,
-          "coef_reg_cnn": 1e-4,
-          "coef_reg_den": 1e-4,
-          "dropout_rate": 0.5,
-          "dense_size": 100,
-          "model_name": "cnn_model"
-        },
-        "embedder": {
-          "name": "fasttext",
-          "save_path": "embeddings/dstc2_fastText_model.bin",
-          "load_path": "embeddings/dstc2_fastText_model.bin",
-          "emb_module": "fasttext",
-          "dim": 100
-        },
-        "tokenizer": {
-          "name": "nltk_tokenizer",
-          "tokenizer": "wordpunct_tokenize"
-        }
+        "kernel_sizes_cnn": [
+          1,
+          2,
+          3
+        ],
+        "filters_cnn": 512,
+        "confident_threshold": 0.5,
+        "optimizer": "Adam",
+        "lear_rate": 0.1,
+        "lear_rate_decay": 0.1,
+        "loss": "binary_crossentropy",
+        "text_size": 15,
+        "coef_reg_cnn": 1e-4,
+        "coef_reg_den": 1e-4,
+        "dropout_rate": 0.5,
+        "dense_size": 100,
+        "model_name": "cnn_model",
+        "embedder": "#my_embedder",
+        "tokenizer": "#my_tokenizer"
       }
     ],
-    "out": ["y_predicted"]
+    "out": [
+      "y_labels",
+      "y_probas",
+      "y_classes"
```

Comment:
It's weird that classes are returned as a model's output in every batch. Will it be bad if probabilities will be returned in the form of `dict(zip(y_classes, y_probas))`?

Reply:
done